src/bot/precompile_include.jl

```diff
@@ -0,0 +1,144 @@
+"""
+    detectOS()
+
+Returns Operating System of a machine as a string as the 1st output and the osfun as the 2nd output.
+"""
+function detectOS()
+allos_funs = [
+         Sys.iswindows,
+         Sys.isapple,
+         Sys.islinux,
+         Sys.isbsd,
+         Sys.isdragonfly,
+         Sys.isfreebsd,
+         Sys.isnetbsd,
+         Sys.isopenbsd,
+         Sys.isjsvm]
+    os = ""
+    osfun = allos_funs[1] # temp
+    for osfun in allos_funs
+        if osfun()
+            os = string(osfun)[3:end]
+            break
+        end
+    end
+    if os == ""
+        @error "os is not detected"
+    end
+    return os, osfun
+end
+################################################################
+"""
+    new_includer_file(package_name::String, package_path::String, os::Union{Vector{String}, Nothing})
+
+Creates a "precompile_includer.jl" file.
+"""
+function new_includer_file(package_name::String, package_path::String, os::Union{Vector{String}, Nothing})
+    includer_file = joinpath(dirname(package_path), "precompile_includer.jl")
+
+    if isnothing(os)
+        multiosstr = ""
+        ismultios = false
+    else
+        multiosstr = ""
+        for eachos in os
+            multiosstr = multiosstr * """elseif Sys.is$eachos()
+                include("../deps/SnoopCompile/precompile/$eachos/precompile_$package_name.jl")
+                _precompile_()
+            """
+        end
+        ismultios = true
+    end
+
+    @info "$includer_file file will be created/overwritten"
+    enclusure = """
```

Comment:
The word "enclusure" seems a bit unusual, what does it mean? Google isn't helping much.

Reply:
Sorry! That's a typo. 😄